packages/aws-cdk-lib/aws-codebuild/lib/fleet.ts

```diff
@@ -125,6 +191,7 @@ export class Fleet extends Resource implements IFleet {
   public readonly environmentType: EnvironmentType;
 
   constructor(scope: Construct, id: string, props: FleetProps) {
+    super(scope, id, { physicalName: props.fleetName });
```

Comment:
I just noticed this addition, thanks I'd missed the `physicalName` in my original implementation. Wouldn't something like this be preferable though, I don't see a reason not to give `Resource` the rest of the props?
```ts
super(scope, id, { ...props, physicalName: props.fleetName });
```